app/models/authorization.server.ts

```diff
@@ -0,0 +1,65 @@
+import { isProjectTeamMember } from "./project.server";
+
+export type Roles =
+  | "ADMIN"
+  | "USER"
+  | "APPLICANT"
+  | "ANONYMOUS"
+  | "project-team-member";
+
+type Actions = "create" | "view" | "edit" | "delete" | "edit.project";
+
+type Resources = "applicant" | "project";
+
+export function checkPermission(
+  principal: string | number,
+  role: Roles,
+  action: Actions,
+  resourceType: Resources,
+  resource: any
+) {
```

Comment:
This is inspired in https://github.com/cerbos/cerbos concepts. Specifically "Key concepts, at a glance 👀"

cc @LuisTejedaS